### PR TITLE
Fixed bug #5637

### PR DESCRIPTION
**Pull request by whyb**
Fixed bug #5637

https://github.com/Tencent/ncnn/issues/5637

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=5640) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=5640) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=5640) it.</sub>

File: examples/yolov8.cpp [hunk 1]
```diff
@@ -175,10 +175,10 @@ static void parse_yolov8_detections(
...
     for (int i = 0; i < num_anchors; i++)
     {
-        auto row_ptr = output.row(i).ptr<float>();
-        auto bboxes_ptr = row_ptr;
-        auto scores_ptr = row_ptr + 4;
-        auto max_s_ptr = std::max_element(scores_ptr, scores_ptr + num_labels);
+        float* row_ptr = output.row(i).ptr<float>();
+        float* bboxes_ptr = row_ptr;
+        float* scores_ptr = row_ptr + 4;
+        float* max_s_ptr = std::max_element(scores_ptr, scores_ptr + num_labels);
```
Comment (nihui) [inline examples/yolov8.cpp:181]:
```suggestion
        const float* row_ptr = output.row(i);
        const float* bboxes_ptr = row_ptr;
        const float* scores_ptr = row_ptr + 4;
        const float* max_s_ptr = std::max_element(scores_ptr, scores_ptr + num_labels);
```

File: examples/yolov8.cpp [hunk 2]
```diff
@@ -175,10 +175,10 @@ static void parse_yolov8_detections(
 
     for (int i = 0; i < num_anchors; i++)
     {
-        auto row_ptr = output.row(i).ptr<float>();
-        auto bboxes_ptr = row_ptr;
-        auto scores_ptr = row_ptr + 4;
-        auto max_s_ptr = std::max_element(scores_ptr, scores_ptr + num_labels);
+        const float* row_ptr = output.row(i);
```
Comment (nihui) [inline examples/yolov8.cpp:178]:
```suggestion
        const float* row_ptr = output.row(i).ptr<float>();
```

Comment (nihui) [reply examples/yolov8.cpp:178]: 看错了，原来是 cv::Mat ...

**Comment by nihui** [comment]
Thanks for your contribution !

**Pull request merged**